### PR TITLE
Android: Catch IllegalStateException on local file playback

### PR DESCRIPTION
**Pull request by benvium**

### CODE REVIEW

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 1]
```diff
@@ -87,7 +88,13 @@ public synchronized boolean onError(MediaPlayer mp, int what, int extra) {
         return true;
       }
     });
-    player.prepareAsync();
+
+    try {
+      player.prepareAsync();
+    } catch (IllegalStateException ignored) {
+      // When loading files from a file, we useMediaPlayer.create, which actually
```
Comment (fabiancook) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: Would this not mean we never invoke our callback? Can we detect the state?

Comment (benvium) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: AFAIK that exception is only thrown if the preparation has already been done. Actual errors should still happen.

Comment (fabiancook) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: Yes, but `callback` would never be invoked, which means react native is never informed. Unless onPrepared is invoked each time?

Comment (benvium) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: ... and detecting the state isn't possible. There's no 'isPrepared' or 'getState' on MediaPlayer :-(

Comment (fabiancook) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: If onPrepared isn't invoked, we could split the creation to a file based media player, and an http/s based one

**Pull request merged**